internal/retry/retry.go

```diff
@@ -104,7 +119,7 @@ func (r *OutboundMiddleware) Call(ctx context.Context, request *transport.Reques
 			return resp, err
 		}
 
-		// TODO add backoff semantics
+		time.Sleep(r.getBackoff(i))
```

Comment:
General thing for all of this: it would be nice to not use the global `time.Sleep` in the code, but instead have an interface such as:

```go
type Sleeper func(time.Duration)
```

Where you could switch out the Sleeper, especially for testing.